addons/sales_team/models/crm_team.py

```diff
@@ -26,16 +34,213 @@ def _get_default_team_id(self, user_id=None):
+        locale = self._context.get('lang', 'en_US')
+        if self.kanban_dashboard_graph_group == 'day':
+            for day in range(0, (end_date - start_date).days + 1):
+                # name = format_date(start_date, 'd LLLL Y', locale=locale)
+                short_name = format_date(start_date + relativedelta(days=day), 'd MMM', locale=locale)
+                values.append({x_field: short_name, y_field: 0}) # , 'name': name
+            for data_item in graph_data:
+                index = (datetime.strptime(data_item.get('x_value'), DF).date() - start_date).days
+                values[index][y_field] = data_item.get('y_value')
+
+        elif self.kanban_dashboard_graph_group == 'week':
+            for week in range(0, (end_date.isocalendar()[1] - start_date.isocalendar()[1]) % 52 + 1):
+                short_name = get_week_name(start_date + relativedelta(days=7 * week), locale)
+                values.append({x_field: short_name, y_field: 0}) # 'name': name
+
+            for data_item in graph_data:
+                # FIXME: once every 28 years, this will mess up :/ and we'll have 2 values for a same x_value (2nd one will override), and none for the last
+                # why ? leap years, 1 every 7 will have 53 weeks -> on the next year 53-53%52=0, and 1-53%52=0
+                # max period is a year so no problem there
+                index = int((data_item.get('x_value') - start_date.isocalendar()[1]) % 52)
+                values[index][y_field] = int(data_item.get('y_value'))
+
+        elif self.kanban_dashboard_graph_group == 'month':
+            for month in range(0, (end_date.month - start_date.month) % 12 + 1):
+                short_name = format_date(start_date + relativedelta(months=month), 'MMM', locale=locale)
+                values.append({x_field: short_name, y_field: 0}) # 'name': name
+
+            for data_item in graph_data:
+                index = int((data_item.get('x_value') - start_date.month) % 12)
+                values[index][y_field] = data_item.get('y_value')
+
+        elif self.kanban_dashboard_graph_group == 'user':
+            for data_item in graph_data:
+                values.append({x_field: self.env['res.users'].browse(data_item.get('x_value')).name, y_field: data_item.get('y_value')})
+
+        [graph_title, graph_key] = self._graph_title_and_key()
+        return [{'values': values, 'area': True, 'title': graph_title, 'key': graph_key}]
+
+    def _compute_dashboard_button_name(self):
+        """ Sets the adequate dashboard button name depending on the sales channel's options
+        """
+        for team in self:
+            team.dashboard_button_name = _("Big Pretty Button :)") # placeholder
+
+    def action_primary_channel_button(self):
+        """ skeleton function to be overloaded
+            It will return the adequate action depending on the sales channel's options
+        """
+        return
+
+    @api.model
+    def _verify_team_type(self, values):
```

Comment:
Yes exactly, if CRM is installed, use_opportunities defaults to true (using a pipeline with leads/opportunities).
If I create a new sales channel of type point of sale, I will need to have use_opportunities set to False no matter what. Otherwise I'd enable actions that make no sense from the dashboard.